client/network/src/protocol.rs

```diff
@@ -1434,24 +1434,11 @@ impl<B: BlockT, H: ExHashT> Protocol<B, H> {
 		}
 	}
 
-	/// Call this when a block has been imported in the import queue and we should announce it on
-	/// the network.
-	pub fn on_block_imported(&mut self, header: &B::Header, data: Vec<u8>, is_best: bool, also_announce: bool) {
+	/// Call this when a block has been imported in the import queue
+	pub fn on_block_imported(&mut self, header: &B::Header, is_best: bool) {
```

Comment:
TBH, seems that this function now does nothing unless the new block is best. Maybe it should be `fn on_best_block_imported(&mut self, header: &B::Header)`

Reply:
But then we would need to reintroduce it if later we want to do something for non-best blocks.

I quite dislike that function to be honest, but its API seems to be "please call me when a block has been added to the client", and adding an "only if it is the new best block" clause seems more error-prone.